src/components/TransactionSettingsBottomSheet.tsx

```diff
@@ -1,173 +1,534 @@
+  };
 
-  /**
-   * Handles confirmation of transaction settings
-   * Validates all inputs and saves to global store if valid
-   */
   const handleConfirm = () => {
-    if (memoError || feeError || timeoutError) return;
-    saveMemo(localMemo);
-    saveTransactionTimeout(Number(localTimeout));
-    saveTransactionFee(localFee);
+    const hasErrors = settings.some((setting) => settingErrors[setting]);
+    if (hasErrors) return;
+
+    settings.forEach((setting) => {
+      settingSaveCallbacks[setting]();
+    });
+
+    // Notify that settings have changed
+    onSettingsChange?.();
+
     onConfirm();
   };
 
-  /**
-   * Converts network congestion level to localized string
-   *
-   * @param {NetworkCongestion} congestion - The network congestion level
-   * @returns {string} Localized string representation of congestion level
-   */
-  const getLocalizedCongestionLevel = (
-    congestion: NetworkCongestion,
-  ): string => {
-    switch (congestion) {
-      case NetworkCongestion.LOW:
-        return t("low");
-      case NetworkCongestion.MEDIUM:
-        return t("medium");
-      case NetworkCongestion.HIGH:
-        return t("high");
-      default:
-        return t("low");
-    }
-  };
+  // Render functions
+  const getMemoRow = useCallback(
+    () => (
+      <View className="flex-col gap-2 mt-[24px]">
+        <View className="flex flex-row items-center gap-2">
+          <Text sm secondary>
+            {t("transactionSettings.memoTitle")}
+          </Text>
+          <TouchableOpacity
+            onPress={() => memoInfoBottomSheetModalRef.current?.present()}
+          >
+            <Icon.InfoCircle color={themeColors.gray[8]} size={pxValue(16)} />
+          </TouchableOpacity>
+        </View>
+        <Input
+          isBottomSheetInput
+          fieldSize="lg"
+          leftElement={
+            <Icon.File02
+              size={pxValue(16)}
+              color={themeColors.foreground.primary}
+            />
+          }
+          placeholder={t("transactionSettings.memoPlaceholder")}
+          value={localMemo}
+          onChangeText={handleMemoChange}
+          error={memoError}
+        />
+      </View>
+    ),
+    [
+      localMemo,
+      memoError,
+      t,
+      themeColors.foreground.primary,
+      themeColors.gray,
+      handleMemoChange,
+    ],
+  );
+
+  const getSlippageRow = useCallback(
+    () => (
+      <View className="flex-col gap-2 mt-[24px]">
+        <View className="flex flex-row items-center justify-between">
+          <View className="flex flex-row items-center gap-2">
+            <Text sm secondary>
+              {t("transactionSettings.slippageTitle")}
+            </Text>
+            <TouchableOpacity
+              onPress={() => slippageInfoBottomSheetModalRef.current?.present()}
+            >
+              <Icon.InfoCircle color={themeColors.gray[8]} size={pxValue(16)} />
+            </TouchableOpacity>
```

Comment:
```suggestion
              <Icon.InfoCircle color={themeColors.gray[8]} size={16} onPress={() => slippageInfoBottomSheetModalRef.current?.present()} />
```
can you also check if you cant use the `themeColor="gray"` here? it will render the `themeColors.gray[9]` instead of `themeColors.gray[8]`, worth checking with @sdfcharles